### PR TITLE
Add prelude

### PR DESCRIPTION
**Pull request by andy-thomason**
This PR adds a prelude to extendr_api.

From now on, use:

```
use extendr_api::prelude::*;
```
and not
```
use extendr_api::*;
```
The latter will be turned off in the next few weeks.

The tests have been updated to use the test! macro and the prelude.

### CODE REVIEW

**Review by clauswilke — APPROVED** [review]

**Comment by clauswilke** [comment]
There are still some problems with the integration tests, but otherwise looks good to me.

**Comment by Ilia-Kosenkov** [comment]
There is some issue with R-devel for Windows, I saw this earlier in my fork as well:
https://github.com/extendr/extendr/pull/156/checks?check_run_id=1780350499#step:4:15
Perhaps something happened during an update to R actions, I suggest to temporarily roll back to release.

**Comment by yutannihilation** [comment]
It seems it's not only us seeing this failure: https://github.com/r-lib/actions/issues/234

File: extendr-api/src/functions.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use crate::*;
 ///
 /// Use the Env{} wrapper for more detail.
 /// ```
-/// use extendr_api::*;
+/// use extendr_api::prelude::*;;
```
Comment (yutannihilation) [inline extendr-api/src/functions.rs:7]: This file contains some double `;`. Maybe you mistyped a regex replacement?

Comment (andy-thomason) [reply extendr-api/src/functions.rs:7]: Very likely!

**Review by yutannihilation — APPROVED** [review]

**Pull request merged**